src/App.tsx

```diff
@@ -1,14 +1,65 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterType } from './types/FilterType';
+
+export function filterTodos(
+  todos: Todo[],
+  statusSelect: FilterType,
+  searchResult: string,
+): Todo[] {
+  let prepareTodos = [...todos];
+
+  if (searchResult) {
+    prepareTodos = prepareTodos.filter(todo => {
+      return todo.title
+        .toLowerCase()
+        .includes(searchResult.toLowerCase().trim());
+    });
+  }
+
+  switch (statusSelect) {
+    case FilterType.COMPLETED:
+      prepareTodos = prepareTodos.filter(todo => todo.completed);
+      break;
+    case FilterType.ACTIVE:
+      prepareTodos = prepareTodos.filter(todo => !todo.completed);
+      break;
+    default:
+      break;
+  }
+
+  return prepareTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [statusSelect, setStatusSelect] = useState(FilterType.ALL);
+  const [searchResult, setSearchResult] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      });
+  }, []);
+
+  const onShowModal = (todo: Todo) => setSelectedTodo(todo);
+  const onCloseModal = () => setSelectedTodo(null);
```

Comment:
```suggestion
  const handleShowModal = (todo: Todo) => setSelectedTodo(todo);
  const handleCloseModal = () => setSelectedTodo(null);
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364